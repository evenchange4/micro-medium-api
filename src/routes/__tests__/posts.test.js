/* eslint global-require: 0, no-throw-literal: 0 */

beforeEach(() => {
  jest.resetModules();
  jest.mock('micro', () => ({
    send: (res, code, content) => ({ res, code, content }),
  }));
  jest.mock('../../utils/toJSON', () => d => d);
  jest.mock('../../utils/simplify', () => () => d => d);
});

it('should return posts', async () => {
  jest.doMock('../../utils/API', () => ({
    getPosts: (username, query) => ({
      username,
      query,
    }),
  }));
  const posts = require('../posts');

  expect(
    await posts({ query: {}, params: { username: 'username' } }),
  ).toMatchSnapshot();
});

it('should return posts with simplify', async () => {
  jest.doMock('../../utils/API', () => ({
    getPosts: (username, query) => ({
      username,
      query,
    }),
  }));
  const posts = require('../posts');
  expect(
    await posts({
      query: { type: 'simple' },
      params: { username: 'username' },
    }),
  ).toMatchSnapshot();
});

it('should return posts with error', async () => {
  jest.doMock('../../utils/API', () => ({
    getPosts: (username, query) => {
      throw { response: { username, query, test: 'error response' } };
    },
  }));
  const posts = require('../posts');

  expect(
    await posts({
      query: { type: 'simple' },
      params: { username: 'username' },
    }),
  ).toMatchSnapshot();
});
