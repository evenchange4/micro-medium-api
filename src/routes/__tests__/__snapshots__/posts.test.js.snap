// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return posts 1`] = `
Object {
  "code": 200,
  "content": Object {
    "query": Object {},
    "username": "username",
  },
  "res": undefined,
}
`;

exports[`should return posts with error 1`] = `
Object {
  "code": 404,
  "content": Object {
    "query": Object {
      "type": "simple",
    },
    "test": "error response",
    "username": "username",
  },
  "res": undefined,
}
`;

exports[`should return posts with simplify 1`] = `
Object {
  "code": 200,
  "content": Object {
    "query": Object {
      "type": "simple",
    },
    "username": "username",
  },
  "res": undefined,
}
`;
